### PR TITLE
PHP8 compatibility fix

### LINKED ISSUE #10: Error when enabling in RC 12.1.1 and PHP 8.1.1
Greg reports 

> In REDCap 12.1.1 and PHP 8.1.1 the Modify Contact Admin Button throws a fatal error when it is being enabled.
>
> The error message is
> 
> Declaration of Modify_Contact_Admin_Button ($project_id) must be compatible with
> ExternalModules\AbstractModule::redcap_module_configure_button_display() on line 16 of ExternalModule.php.

Here's the entire message and some remediation:
> From: Neils, Gregory (CDC/DDID/NCEZID/OD) (CTR) <XXXX@cdc.gov>
> Date: Tuesday, January 18, 2022 at 5:24 PM
> To: Chase,Philip B <XXXXX@ufl.edu>, Stoffs,Taryn L <XXXXX@ufl.edu>, XXXXX@rub.de <XXXX@rub.de>
> Subject: RE: REDCap EM: Fatal Error Modify Contact Admin Button
> [External Email] 
> 
> ps. Removing the parameter $project_id from the method redcap_module_configure_button_display($projectd_id) fixed it on my local machine. 
> It does not look like there are any adverse effects of removing it either. 
>  
> This is for the REDCap EM Modify Contact Admin Button.  It is the first method in ExternalModule.php on line 16. 
>  
> Greg
>  
> From: Neils, Gregory (CDC/DDID/NCEZID/OD) (CTR) 
> Sent: Tuesday, January 18, 2022 2:21 PM
> To: XXXX@ufl.edu; XXXX@ufl.edu; XXXX@rub.de
> Subject: REDCap EM: Fatal Error Modify Contact Admin Button
>  
> Hi all,
>  
> In REDCap 12.1.1 and PHP 8.1.1 the Modify Contact Admin Button throws a fatal error when it is being enabled. 
>  
> The error message is
>  
> Declaration of Modify_Contact_Admin_Button ($project_id) must be compatible with 
> ExternalModules\AbstractModule::redcap_module_configure_button_display() on line 16 of ExternalModule.php. Stack trace is not available. 
>  
> Let me know if I can help debug this in any way. 
>  
> Thanks all

### PR DESCRIPTION
**Pull request by grezniczek**
@pbchase , this should fix #10.

Other than that tiny little change, I don't really see any issues with PHP8 (running this module on my dev env PHP 8.1.2)

### CODE REVIEW

**Review by michael-bentz — APPROVED** [review]
To get the proper environment with REDCap 12.1.1 and PHP 8.1 I used [redcap docker compose](https://github.com/123andy/redcap-docker-compose) with two edits to the docker-web `Dockerfile`:
- Changed based image from `php:7.3-apache` to `php:8.1-apache`
- Commented out `docker-php-ext-configure zip --with-libzip`

The change fixes the issue for PHP 8.1 and is backwards compatible to version 8.0.3.

**Comment by michael-bentz** [comment]
@pbchase Do you mind the changelog updates being part of this PR?

**Comment by grezniczek** [comment]
@mbentz-uf , @pbchase - I noticed two more places throwing in PHP in some contexts (undefined constants).

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 All notable changes to the Modify Contact Admin Button project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.1.1] - 2022-03-07
+### Changed
+- PHP8 compatibility fixes (GR)
+
```
Comment (pbchase) [inline CHANGELOG.md]: Let's strike this part. We write ChangeLog as part of the release workflow. The data and sometimes even the version number are unknown until the moment of release. Pre-drafting the content interferes with the scripts we use to do the releases.

```suggestion
```

**Pull request merged**